Fix doxygen param formatting

owen-mc-diffblue opened the pull request:
Fix formatting errors in doxygen function comments. I've tried to keep things in the right commits, but some might have slipped through.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [n/a] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [n/a] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [n/a] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [n/a] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [n/a] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

Commit:
Standardise formatting of doxygen for out parameters

Use "[in,out]" instead of "[in/out]" for in-and-out-parameters.

Make sure "[out]" and "[in,out]" come between "\param" and the
parameter name.

Commit:
Clarify doxygen format for in-out-parameters

Commit:
Fix "\param arg :" in doxygen

Commit:
Add colons after parameter name in doxygen

Commit:
Miscellaneous fixes to doxygen parameters

Commit:
Single space after "\return" in doxygen

Commit:
Only one space after "\param x:" in doxygen

Commit:
Fix indentation in doxygen

Commit:
Fix lines beginning with // instead of ///

tautschnig commented on jbmc/src/java_bytecode/java_bytecode_convert_class.cpp:
```diff
@@ -1036,7 +1036,7 @@ static void find_and_replace_parameter(
 /// \param type
 /// \param replacement_parameters
-static void find_and_replace_parameters(
+static: void find_and_replace_parameters(
```
It seems the scripting got something wrong here, which is caused by the missing documentation of the parameters.

tautschnig commented on jbmc/src/java_bytecode/java_bytecode_convert_class.cpp:
```diff
@@ -1120,7 +1120,7 @@ void convert_java_annotations(
 /// \param class_name
 /// \param symbol_table
-void mark_java_implicitly_generic_class_type(
+void: mark_java_implicitly_generic_class_type(
```
As above: documentation missing

tautschnig reviewed (approved):
Thanks a lot for all this cleanup work! The below changes are strictly required, but CI will tell you anyway and thus approving right away.

Commit:
Remove extra lines between functions

Clang-format was complaining about them because the previous commit
fixing indentation changed the function comments

owen-mc-diffblue commented:
I applied the fixes for @tautschnig's comments (I should have been more careful with my regex), removed some unneeded empty lines between functions that clang-format was complaining about and some overly long lines that the linter was complaining about. It now passes all the quick checks, so it should be good to go.

Commit:
Indent continuation lines after \param and \return in doxygen

Commit:
Replace \returns with \return in doxygen

Commit:
Remove colon after \return in doxygen

Commit:
Do not use @param and @return in doxygen

owen-mc-diffblue commented:
@tautschnig I added four more commits, because I noticed some more easily fixable formatting problems

allredj reviewed (commented):
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 8d2912dc2b5e60c2b9003c9d8fae3fd485f09607).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95618858
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 7c364c1e93b577e7c6d65c9f3286e17ff9ebc232).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95619117

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1957e4d7c225553c67eebea9c91b61041a02a48e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95621342

tautschnig commented on jbmc/src/java_bytecode/remove_exceptions.cpp:
```diff
@@ -260,13 +260,13 @@ void remove_exceptionst::instrument_exception_handler(
+/// \param[out] universal_try returns the try block corresponding to the
 ///   desired exception handler
-/// @param[out] universal_catch returns the catch block corresponding to the
+/// \param[out] universal_catch returns the catch block corresponding to the
```
Should a space before `[out]` be added as well?

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 94180e8959723097ff9b53098391bb00af6fc762).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95752931

owen-mc-diffblue commented:
@cesaro @smowton @thk123 As code owners, can you review this PR, which should be uncontroversial? Then I will fix up the conflicts and merge when CI passes.

smowton reviewed (approved):
Only reviewed insofar as I've scrolled past it at high speed and it looks like all the changes are small and not in code!

owen-mc-diffblue commented:
I can confirm the changes are only in function comments, and are fairly small.

cesaro reviewed (approved):
Looks good to me!

chrisr-diffblue reviewed (approved):
LGTM

Commit:
Remove undocumented parameters

Commit:
Put parameter documentation on the same line as \param

Commit:
Update doxygen to /// instead of /*!

Commit:
Add missing parameter documentation

Commit:
Use \p instead of \param to indicate param name

allredj reviewed (commented):
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 0437520c842aa456ee4148799cf1e88e1f593546).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96180350
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

owen-mc-diffblue commented:
I thought I had enough review now, but apparently not. It's a bit opaque to me, but it seems I need @martin-cs, @peterschrammel and/or @romainbrenguier.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 375b680440652bc85a4cdab18324e5562828642e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96205116
